lib/phoenix_live_view/utils.ex

```diff
@@ -67,7 +67,7 @@ defmodule Phoenix.LiveView.Utils do
   Renders the view with socket into a rendered struct.
   """
   def to_rendered(socket, view) do
-    assigns = Map.put(socket.assigns, :socket, socket)
+    assigns = Map.merge(socket.assigns, %{socket: socket})
```

Comment:
You keep rewriting it as merge and I rewrite it to Map.put. :P

```suggestion
    assigns = Map.put(socket.assigns, :socket, socket)
```